src/provider.js

```diff
@@ -76,14 +76,12 @@ const exec = curry((request, { noAckStream, timeout }, query) => new Promise(
         // reject query on timeout
         // NB: timeout is set in Promise context only when noAckStream
         //     is false and is cancelled in request callback
-        tap(partial(clearTimeout, [ !noAckStream
-          ? setTimeout(
-            partial(reject, [
-              new ProviderError(`query timeout after ${timeout}ms`, query)
-            ]),
-            timeout
-          )
-          : null
+        tap(partial(clearTimeout, [ setTimeout(
```

Comment:
Is there a scenario where `noAckStream` causes a hang without a timeout? The idea is to publish instead of request, so timeout shouldn't be necessary.

Reply:
I don't think this code is ever reached if noAckStream is true, due to the if statement in line 73.

Reply:
> The reason why `noAckStream` is passed to `_streamCreate` that is used in implementing `Writable` part of `Duplex` and not passed to `_create` that is used in `create()`, is because it should only apply to streams. Hence the name. Perhaps I'm missing something here. Do you also want no-ack create?

Ok clear. I will refactor to just pipe to the stream! Thanks for the quick reply 👍  I hope you're doing well!

Reply:
> I don't think this code is ever reached if noAckStream is true, due to the if statement in line 73.

Looks like you're right. If you could amend the PR to exclude passing `noAckStream` to `_create()` this can be merged.

Reply:
Done! Thanks for the super quick actions. It seems to work fine as a no ack stream :)